src/main/kotlin/com/github/treepat/target_tree/default_tree/DefaultTargetTree.kt

```diff
@@ -9,19 +9,26 @@ import org.antlr.v4.runtime.CharStreams
 import org.antlr.v4.runtime.CommonTokenStream
 import org.antlr.v4.runtime.tree.ParseTree
 
-class DefaultTargetTree<T : TargetTreeNode>(override var root: T? = null) :
-    TargetTree<T> {
```

Comment:
I'm not sure about the use of Templates in this situation. I understand that this way you can define which type of `nodes` you want with which type of `trees`. However it would make sense that the `DefaultTargetTree` used `DefaultTargetTreeNodes`. 

And everything is meaningless when we could create the `DefaultTargetTree` only using `TargetTreeNode` (the interface) so I can't see any reason to use a template.

## I got rid of it, but should we keep it? why?